src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EventPublisherContainer.java

```diff
@@ -27,15 +30,24 @@
         this.exceptionRecordCreator = exceptionRecordCreator;
     }
 
-    public EventPublisher getPublisher(Envelope envelope, CaseDetails caseDetails) {
-        switch (envelope.classification) {
+    @NotNull
+    public EventPublisher getPublisher(
+        Classification envelopeClassification,
+        Supplier<CaseDetails> caseRetrieval
+    ) {
+        switch (envelopeClassification) {
             case SUPPLEMENTARY_EVIDENCE:
-                return caseDetails == null ? exceptionRecordCreator : attachDocsPublisher;
+                CaseDetails caseDetails = caseRetrieval.get();
+
+                return caseDetails == null ? exceptionRecordCreator : new DelegatePublisher(
+                    attachDocsPublisher,
+                    caseDetails
+                );
             case EXCEPTION:
                 return exceptionRecordCreator;
             case NEW_APPLICATION:
             default:
-                return null;
+                return new DelegatePublisher(null, null);
```

Comment:
Can we not throw an exception here ? Ideally we are validating the classification in processor anyways so we don't expect this control to be executed.

Reply:
We are. This way is backwards compatible too. We do nothing if we received something unrecognisable. Exception throw will be something new entirely and will bring message back to deadletterqueue. If that's expected - happy to invent one

Reply:
We should at least log in app insights and configure alerts so that bsp team knows whenever this happens in production. Please create a ticket in backlog for alerts config and as of now just log it in app insights. Makes sense ?

Reply:
It is logged. Check out DelegatePublisher ;). Perhaps needs better logging as you said - in separate ticket. A matter of review